File: HackerRank 30 Days/day15/main.cs
using System;

class Node
{
    public int data;
    public Node next;

    public Node(int d)
    {
        data = d;
        next = null;
    }
}

class Solution
{
    public Node insert(Node head, int data)
    {
        Node p = new Node(data);
        if (head == null)
        {
            head = p;
        }
        else
        {
            Node current = head;
            while (current.next != null)
            {
                current = current.next;
            }
            current.next = p;
        }
        return head;
    }

    public void display(Node head)
    {
        Node current = head;
        while (current != null)
        {
            Console.Write(current.data + " ");
            current = current.next;
        }
    }

    static void Main(String[] args)
    {
        Solution mylist = new Solution();
        int T = int.Parse(Console.ReadLine());
        Node head = null;

        for (int i = 0; i < T; i++)
        {
            int data = int.Parse(Console.ReadLine());
            head = mylist.insert(head, data);
        }

        mylist.display(head);
    }
}

File: HackerRank 30 Days/day15/main.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

class Solution {
    insert(head, data) {
        const p = new Node(data);
        if (head === null) {
            head = p;
        } else {
            let current = head;
            while (current.next !== null) {
                current = current.next;
            }
            current.next = p;
        }
        return head;
    }

    display(head) {
        let current = head;
        while (current !== null) {
            process.stdout.write(current.data + ' ');
            current = current.next;
        }
    }
}

function processData(input) {
    const data = input.split('\n');
    const t = parseInt(data[0], 10);
    let head = null;
    const mylist = new Solution();

    for (let i = 1; i <= t; i++) {
        const value = parseInt(data[i], 10);
        head = mylist.insert(head, value);
    }

    mylist.display(head);
}

process.stdin.resume();
process.stdin.setEncoding('ascii');
let _input = '';
process.stdin.on('data', function (input) {
    _input += input;
});
process.stdin.on('end', function () {
    processData(_input);
});
